### PR TITLE
[cluster-test] Add --invalid-tx to support mix txs

### PR DESCRIPTION
**Pull request by snakech**
Added new arg --invalid-tx <num> as percentage of invalid txs(invalid_sender, invalid_receiver, invalid_chain_id)mix in one request batch, this new feature can make longevity test closer to daily usage of chain.

## Motivation

(Write your motivation for proposed changes here.)

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

## Test Plan
1. with no flag:
"/private/var/folders/qd/tg50mr0d7k75qvztr_8yk2km0000gn/T/fba871e6e11af157d5952a94e2885422/mint.key" --swarm --peers "127.0.0.1:8080" --emit-tx --chain-id TESTING

```
2020-12-16T21:11:25.966953Z [main] INFO testsuite/cluster-test/src/tx_emitter.rs:560 submitted: 60 txn/s, committed: 45 txn/s, expired: 0 txn/s, latency: 2641 ms, p99 latency: 2750 ms
2020-12-16T21:11:35.968835Z [main] INFO testsuite/cluster-test/src/tx_emitter.rs:560 submitted: 45 txn/s, committed: 45 txn/s, expired: 0 txn/s, latency: 2664 ms, p99 latency: 2850 ms
2020-12-16T21:11:45.969676Z [main] INFO testsuite/cluster-test/src/tx_emitter.rs:560 submitted: 60 txn/s, committed: 60 txn/s, expired: 0 txn/s, latency: 2698 ms, p99 latency: 3050 ms
2020-12-16T21:11:55.972634Z [main] INFO testsuite/cluster-test/src/tx_emitter.rs:560 submitted: 46 txn/s, committed: 46 txn/s, expired: 0 txn/s, latency: 2745 ms, p99 latency: 2850 ms
2020-12-16T21:12:05.974936Z [main] INFO testsuite/cluster-test/src/tx_emitter.rs:560 submitted: 58 txn/s, committed: 58 txn/s, expired: 0 txn/s, latency: 2783 ms, p99 latency: 2950 ms
2020-12-16T21:12:15.976401Z [main] INFO testsuite/cluster-test/src/tx_emitter.rs:560 submitted: 46 txn/s, committed: 46 txn/s, expired: 0 txn/s, latency: 2748 ms, p99 latency: 2850 ms
Total stats: submitted: 3165, committed: 3165, expired: 0
Average rate: submitted: 52 txn/s, committed: 52 txn/s, expired: 0 txn/s, latency: 2708 ms, p99 latency: 2950 ms
```

2. with --invalid-tx 0
"/private/var/folders/qd/tg50mr0d7k75qvztr_8yk2km0000gn/T/fba871e6e11af157d5952a94e2885422/mint.key" --swarm --peers "127.0.0.1:8080" --emit-tx --chain-id TESTING --invalid-tx 0

```
2020-12-16T21:11:25.966953Z [main] INFO testsuite/cluster-test/src/tx_emitter.rs:560 submitted: 60 txn/s, committed: 45 txn/s, expired: 0 txn/s, latency: 2641 ms, p99 latency: 2750 ms
2020-12-16T21:11:35.968835Z [main] INFO testsuite/cluster-test/src/tx_emitter.rs:560 submitted: 45 txn/s, committed: 45 txn/s, expired: 0 txn/s, latency: 2664 ms, p99 latency: 2850 ms
2020-12-16T21:11:45.969676Z [main] INFO testsuite/cluster-test/src/tx_emitter.rs:560 submitted: 60 txn/s, committed: 60 txn/s, expired: 0 txn/s, latency: 2698 ms, p99 latency: 3050 ms
2020-12-16T21:11:55.972634Z [main] INFO testsuite/cluster-test/src/tx_emitter.rs:560 submitted: 46 txn/s, committed: 46 txn/s, expired: 0 txn/s, latency: 2745 ms, p99 latency: 2850 ms
2020-12-16T21:12:05.974936Z [main] INFO testsuite/cluster-test/src/tx_emitter.rs:560 submitted: 58 txn/s, committed: 58 txn/s, expired: 0 txn/s, latency: 2783 ms, p99 latency: 2950 ms
2020-12-16T21:12:15.976401Z [main] INFO testsuite/cluster-test/src/tx_emitter.rs:560 submitted: 46 txn/s, committed: 46 txn/s, expired: 0 txn/s, latency: 2748 ms, p99 latency: 2850 ms
Total stats: submitted: 3165, committed: 3165, expired: 0
Average rate: submitted: 52 txn/s, committed: 52 txn/s, expired: 0 txn/s, latency: 2708 ms, p99 latency: 2950 ms
```

3. with --invalid-tx 50
cargo run -p cluster-test -- --mint-file "/private/var/folders/qd/tg50mr0d7k75qvztr_8yk2km0000gn/T/fba871e6e11af157d5952a94e2885422/mint.key" --swarm --peers "127.0.0.1:8080" --emit-tx --chain-id TESTING --invalid-tx 50
```
2020-12-16T21:08:28.569420Z [tokio-runtime-worker] WARN testsuite/cluster-test/src/tx_emitter.rs:668 [http://127.0.0.1:8080/v1] Failed to submit request: InvalidServerResponse("JsonRpcError { code: -32001, message: \"Server error: VM Validation error: SENDING_ACCOUNT_DOES_NOT_EXIST\", data: Some(StatusCode(SENDING_ACCOUNT_DOES_NOT_EXIST)) }")
2020-12-16T21:08:28.576991Z [tokio-runtime-worker] WARN testsuite/cluster-test/src/tx_emitter.rs:668 [http://127.0.0.1:8080/v1] Failed to submit request: InvalidServerResponse("JsonRpcError { code: -32001, message: \"Server error: VM Validation error: SENDING_ACCOUNT_DOES_NOT_EXIST\", data: Some(StatusCode(SENDING_ACCOUNT_DOES_NOT_EXIST)) }")
2020-12-16T21:08:28.584729Z [tokio-runtime-worker] WARN testsuite/cluster-test/src/tx_emitter.rs:668 [http://127.0.0.1:8080/v1] Failed to submit request: InvalidServerResponse("JsonRpcError { code: -32001, message: \"Server error: VM Validation error: BAD_CHAIN_ID\", data: Some(StatusCode(BAD_CHAIN_ID)) }")
2020-12-16T21:08:28.592124Z [tokio-runtime-worker] WARN testsuite/cluster-test/src/tx_emitter.rs:668 [http://127.0.0.1:8080/v1] Failed to submit request: InvalidServerResponse("JsonRpcError { code: -32001, message: \"Server error: VM Validation error: BAD_CHAIN_ID\", data: Some(StatusCode(BAD_CHAIN_ID)) }")
```

```
Total stats: submitted: 150, committed: 80, expired: 70
Average rate: submitted: 2 txn/s, committed: 1 txn/s, expired: 1 txn/s, latency: 80380 ms, p99 latency: 51150 ms
```

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/diem/diem/developers.libra.org, and link to your PR here.)

### CODE REVIEW

File: testsuite/cluster-test/src/tx_emitter.rs [hunk 1]
```diff
@@ -678,15 +682,29 @@ impl SubmissionWorker {
...
+            let chain_id = if invalid_size > 0 {
+                // we use 255 to be a fake chain id for invalid tx
+                invalid_size -= 1;
+                ChainId::new(255)
+            } else {
+                self.chain_id
+            };
             let request =
-                gen_transfer_txn_request(sender, receiver, SEND_AMOUNT, self.chain_id, gas_price);
+                gen_transfer_txn_request(sender, receiver, SEND_AMOUNT, chain_id, gas_price);
```
Comment (sausagee) [inline testsuite/cluster-test/src/tx_emitter.rs:704]: While invalid ChainID works as invalid TXN, but it's not the full picture.  

Can you cover other edge cases of invalid TXNs?  For example, mixing any of the following, gas fees that are out of range, none existent sender and/or receiver accounts on chain, mint.

File: testsuite/cluster-test/src/tx_emitter.rs [hunk 2]
```diff
@@ -678,15 +727,33 @@ impl SubmissionWorker {
...
+        };
         for sender in accounts {
             let receiver = self
                 .all_addresses
                 .choose(&mut rng)
                 .expect("all_addresses can't be empty");
-            let request =
-                gen_transfer_txn_request(sender, receiver, SEND_AMOUNT, self.chain_id, gas_price);
-            requests.push(request);
+            if invalid_size > 0 {
```
Comment (sausagee) [inline testsuite/cluster-test/src/tx_emitter.rs:741]: This would batch the bad txns together. Randomly mixing/interleaving would have been better.  Anyhow, I think this is good enough to get started.

**Review by sausagee — APPROVED** [review]

File: testsuite/cluster-test/src/tx_emitter.rs (hunk 2 above)
Comment (snakech) [reply testsuite/cluster-test/src/tx_emitter.rs:741]: I think u are correct, I can do a shuffle after we initialized all requests.
